### DISCUSSION TITLE
Support jpackage output

### DISCUSSION THREAD

**Discussion by vogella** [eclipse-tycho/tycho#1022] [Ideas]
As Tycho moved to Java 17, it might be an option to support jpackage for products. 

Jpackage seems to be the new standard for packaging Java applications introduce with Java 14. See https://www.baeldung.com/java14-jpackage

This way the Eclipse SDK could provide Java native installers and the Oomph installer might also be simplified with this. 
cc @merks

**Comment by laeubi** [comment]
@vogella do you have a running example to package an eclipse (equinox) app with `jpackage`?

**Reply by vogella** [reply]
> @vogella do you have a running example to package an eclipse (equinox) app with `jpackage`?

No, I believe this currently does not work due to https://github.com/eclipse/tycho/issues/997 but did not validate. But how cool would it be to have this out of the box of a build job? The produced zip files feel not very modern.

**Reply by laeubi** [reply]
> The produced zip files feel not very modern.

It works well to just extract and start as many eclipse as I like, I don't feel I need an installer at all ;-)

So a first perquisite would be a minimal example to show what tycho has to do here (or even collect missing items).

**Reply by vogella** [reply]
> collect

Agree, just wanted to get a first impression if Tycho devs like the idea or not. I assume that would also remove the need for a native Eclipse launcher, which is already planned to be reduced in complexity. See https://github.com/eclipse-equinox/equinox.framework/issues/37

**Comment by merks** [comment]
From what I read quickly the builds need to happen on the target platform, so that's a significant complication...
